modules/nw-pod-network-connectivity-check-object.adoc

```diff
@@ -0,0 +1,162 @@
+// Module included in the following assemblies:
+//
+//
+
+[id="nw-pod-network-connectivity-check-object_{context}"]
+= PodNetworkConnectivityCheck object fields
+
+The `PodNetworkConnectivityCheck` object fields are described in the following tables.
+
+.PodNetworkConnectivityCheck object fields
+[cols="2,1,3a",options="header"]
+|===
+
+|Field|Type|Description
+
+|`metadata.name`
+|`string`
+|The name of the object in the following format: `<source>-to-<target>`. The destination described by `<target>` includes one of following strings:
+
+* `load-balancer-api-external`
+* `load-balancer-api-internal`
+* `kubernetes-apiserver-endpoint`
+* `kubernetes-apiserver-service-cluster`
+* `network-check-target`
+* `openshift-apiserver-endpoint`
+* `openshift-apiserver-service-cluster`
+
+|`metadata.namespace`
+|`string`
+|The namespace that the object is associated with. This value is always `openshift-network-diagnostics`.
+
+|`spec.sourcePod`
+|`string`
+|The name of the pod where the connection check originates, such as `network-check-source-596b4c6566-rgh92`.
+
+|`spec.targetEndpoint`
+|`string`
+|The target of the connection check, such as `api.devcluster.example.com:6443`.
+
+|`spec.tlsClientCert`
+|`object`
+|An object that configures the TLS certificate to use.
```

Comment:
I'm not sure what this means. Will the users understand?

Reply:
@mburke5678, that's an interesting point. What's an object is sort of fluid.

In the technical sense, it is indeed an object data type. So it's a dependent object, as part of the PNCC object, which itself describes the OCP API for this.

So I think it's probably understood what this means. Tables similar to this appear in the various GitHub repos for different OCP components. So it isn't uncommon.

But it's definitely true that this particular object describes on a subset of the PNCC API.

So I'm tempted to leave it as is.

(It's an object instead of a string because it is possible in the future that `spec.tlsClientCert.name` might not be enough to identify the TLS cert; Maybe some additional property is required. So use of an object here is an escape hatch because you can add `spec.tlsClientCert.style` or whatever later, without breaking the API. In YAML, a stanza or section is always an object:

```yaml
spec:
  tlsClientCert:
   prop1: string
   prop2: 123
```

In a YAML annotation, I've been calling these stanzas or I've seen _section_ used as well.)